tests/multi_thread.rs

```diff
@@ -0,0 +1,80 @@
+#![allow(clippy::undocumented_unsafe_blocks)]
```

Comment:
This test was adapted from my tsync test in extrasafe. https://github.com/boustrophedon/extrasafe/blob/master/tests/thread_multi.rs